apps/settings/js/modules/bluetooth/bluetooth_device.js

```diff
@@ -30,14 +30,105 @@ define(function(require) {
    */
   return function ctor_bluetooth_device(device) {
     var type = BtClassOfDeviceMapper.getDeviceType(device.cod);
+    var connectionStatus = device.connectionStatus || 'disconnected';
     var observableBluetoothDevice = Observable({
       name: device.name,
       paired: device.paired,
       address: device.address,
-      type: type
+      type: type,
+      connectionStatus: connectionStatus,
+      profiles: null,
+      descriptionText: '',
+      data: device
     });
 
     /**
+     * Provide a function to update connection info.
+     *
+     * @access public
+     * @memberOf BluetoothDevice
+     * @param {Object} options
+     * @param {Boolean} options.connectionStatus - is connected or not
+     * @param {Object} options.profiles - profiles of the connection type
+     */
+    observableBluetoothDevice.updateConnectionInfo =
```

Comment:
We should declare the function in the outer scope and simply use the reference of it so the same function won't be created every time we create a new bluetooth device.
```
function updateConnectionInfo() {}
return function ctor_bluetooth_device(device) {
  var observableBluetoothDevice = Observable({
    updateConnectionInfo: updateConnectionInfo
  });
};